File: backend/app/models/organization.py
from pydantic import BaseModel
from typing import Optional


class OrganizationCreate(BaseModel):
    cui: str
    company_name: str
    trade_register_nr: Optional[str] = None
    address_street: Optional[str] = None
    address_city: Optional[str] = None
    address_county: Optional[str] = None
    address_postal_code: Optional[str] = None
    vat_registered: bool = False
    bank_account: Optional[str] = None
    bank_name: Optional[str] = None
    invoice_series: str = "FACT"


class OrganizationResponse(BaseModel):
    id: str
    user_id: str
    cui: str
    company_name: str
    trade_register_nr: Optional[str] = None
    address_street: Optional[str] = None
    address_city: Optional[str] = None
    address_county: Optional[str] = None
    address_postal_code: Optional[str] = None
    vat_registered: bool
    bank_account: Optional[str] = None
    bank_name: Optional[str] = None
    invoice_series: str
    next_invoice_number: int
    created_at: str
    updated_at: str

File: frontend/src/types/organization.ts
export interface Organization {
  id: string
  user_id: string
  cui: string
  company_name: string
  trade_register_nr?: string
  address_street?: string
  address_city?: string
  address_county?: string
  address_postal_code?: string
  vat_registered: boolean
  bank_account?: string
  bank_name?: string
  anaf_token_expires_at?: string
  invoice_series: string
  next_invoice_number: number
  created_at: string
  updated_at: string
}
